conanfile.py

```diff
@@ -34,85 +34,38 @@ def configure(self):
-                                  '    <ClCompile Include="..\\..\\src\\opus_compare.c">\n'
-                                  '      <DisableSpecificWarnings>4244;%(DisableSpecificWarnings)'
-                                  '</DisableSpecificWarnings>\n'
-                                  '    </ClCompile>\n', '')
-
-    @property
-    def fixed_point(self):
-        if self.settings.compiler == 'Visual Studio' and not self.options.shared:
-            return False
-        return self.options.fixed_point
-
-    def build_msvc(self):
-        with tools.chdir(self._source_subfolder):
-            pc_build = 'fixed-point' if self.fixed_point else 'floating-point'
-            shutil.copy('opus.pc.in', 'opus.pc')
-            tools.replace_in_file('opus.pc', '@VERSION@', self.version)
-            tools.replace_in_file('opus.pc', '@PC_BUILD@', pc_build)
-            tools.replace_in_file('opus.pc', '@LIBM@', '')
-            tools.replace_in_file('opus.pc', '@prefix@', os.path.abspath(self.package_folder))
-            tools.replace_in_file('opus.pc', '@exec_prefix@', '${prefix}')
-            tools.replace_in_file('opus.pc', '@libdir@', '${exec_prefix}/lib')
-            tools.replace_in_file('opus.pc', '@includedir@', '${prefix}/include')
-        with tools.chdir(os.path.join(self._source_subfolder, "win32", "VS2015")):
-            btype = "%s%s%s" % (self.settings.build_type, "DLL" if self.options.shared else "",
-                                "_fixed" if self.fixed_point else "")
-            msbuild = MSBuild(self)
-            msbuild.build('opus.sln', build_type=btype, platforms={"x86": "Win32"})
-
-    def build_configure(self):
-        def chmod_plus_x(filename):
-            if os.name == 'posix':
-                os.chmod(filename, os.stat(filename).st_mode | 0o111)
+        # They forgot to package that file into the tarball for 1.3.1
+        # See https://github.com/xiph/opus/issues/129
+        os.rename("opus_buildtype.cmake", self._source_subfolder + "/opus_buildtype.cmake")
 
-        with tools.chdir(self._source_subfolder):
-            if self.settings.os == "Macos":
-                tools.replace_in_file("configure", r"-install_name \$rpath/", "-install_name ")
-            chmod_plus_x('configure')
-            if self.options.shared:
-                args = ['--disable-static', '--enable-shared']
-            else:
-                args = ['--disable-shared', '--enable-static']
-            args.append('--enable-fixed-point' if self.fixed_point else '--disable-fixed-point')
-            env_build = AutoToolsBuildEnvironment(self)
-            env_build.configure(args=args)
-            env_build.make()
-            env_build.install()
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["OPUS_FIXED_POINT"] = self.options.fixed_point
+        cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        cmake.configure(source_folder=self._source_subfolder)
```

Comment:
you need to add a CMake wrapper file to call conan_basic_setup.
Copy this file : https://github.com/bincrafters/conan-templates/blob/master/default/CMakeLists.txt and export it by exports_source.
You can take a look in https://github.com/bincrafters/conan-libdwarf as reference. You will see we are using `cmake.configure()` because we consume the cmake wrapper